### PR TITLE
Refactoring to handle Watch command type in the generic flow of command handlers

### PR DESCRIPTION
**Pull request by AshwinKul28**
This update introduces a refactor to incorporate support for the Watch command type within the existing generic command handling flow.

Just one query: I was not sure where to put the `internal/querymanager/watch_response.go` file. Keeping it in the querymanager as it is more relevant. Feel free to move @JyotinderSingh

### CODE REVIEW

**Comment by AshwinKul28** [comment]
Screenshot 2024-10-09 at 9 45 49 PM

Also, I have seen this error even on master as well.

**Comment by JyotinderSingh** [comment]
> Screenshot 2024-10-09 at 9 45 49 PM
> Also, I have seen this error even on master as well.

The command is not supported on CLI yet.

**Comment by JyotinderSingh** [comment]
> Screenshot 2024-10-09 at 9 45 49 PM
> Also, I have seen this error even on master as well.

This may be an issue with our encoding logic

**Comment by AshwinKul28** [comment]
@JyotinderSingh Have a look. once.

**Comment by JyotinderSingh** [comment]
Can you rebase once, looks like the latest changes are not all there

**Comment by AshwinKul28** [comment]
I have sorted those commands as well, hence in the eval->commands it's showing much red. 
All 117 commands are there. @JyotinderSingh

File: internal/worker/worker.go [hunk 1]
```diff
@@ -293,7 +285,7 @@ func (w *BaseWorker) scatter(ctx context.Context, cmds []*cmd.DiceDBCmd) error {
 
 // gather collects the responses from multiple shards and writes the results into the provided buffer.
 // It first waits for responses from all the shards and then processes the result based on the command type (SingleShard, Custom, or Multishard).
-func (w *BaseWorker) gather(ctx context.Context, diceDBCmd *cmd.DiceDBCmd, numCmds, responseType int) error {
+func (w *BaseWorker) gather(ctx context.Context, diceDBCmd *cmd.DiceDBCmd, numCmds int, isWatchChan bool) error {
```
Comment (JyotinderSingh) [inline internal/worker/worker.go:288]:
```suggestion
func (w *BaseWorker) gather(ctx context.Context, diceDBCmd *cmd.DiceDBCmd, numCmds int, isWatchNotification bool) error {
```

File: internal/worker/worker.go [hunk 2]
```diff
@@ -152,8 +155,8 @@ func (w *BaseWorker) Start(ctx context.Context) error {
 	}
 }
 
-func (w *BaseWorker) executeCommandHandler(execCtx context.Context, errChan chan error, cmds []*cmd.DiceDBCmd, isWatchNotification bool) {
-	err := w.executeCommand(execCtx, cmds[0], isWatchNotification)
+func (w *BaseWorker) executeCommandHandler(execCtx context.Context, errChan chan error, cmds []*cmd.DiceDBCmd, isWatchChan bool) {
+	err := w.executeCommand(execCtx, cmds[0], isWatchChan)
```
Comment (JyotinderSingh) [inline internal/worker/worker.go:159]: I feel this name is a little better?
```suggestion
	err := w.executeCommand(execCtx, cmds[0], isWatchNotification)
```

**Review by JyotinderSingh — APPROVED** [review]
LGTM, just a small comment.

File: internal/worker/worker.go (hunk 1 above)
Comment (AshwinKul28) [reply internal/worker/worker.go]: Agreed. done the changes. thanks

**Pull request merged**